app/javascript/components/AvailabilitiesTable.js

```diff
@@ -20,40 +21,52 @@ import { postData } from './utils/sendData';
 import SnackBarComponent from './reusable/SnackBarComponent';
 
 import formatLink from './utils/Link';
+import AvailabilityTimeFormatPrefs from './utils/AvailabilityTimeFormatPrefs';
 
+const styles = {
+  thumbOff: {
+    backgroundColor: '#ffd6cc',
+  },
+  trackOff: {
+    backgroundColor: '#ffb39c',
+  },
+};
 class AvailabilitiesTable extends Component {
   constructor(props, context) {
     super(props, context);
 
     this.handleHideSnackBar = this.handleHideSnackBar.bind(this);
+    this.handleTimeFormatToggle = this.handleTimeFormatToggle.bind(this);
 
     this.state = {
       showSnackBar: false,
-      message: ''
+      message: '',
+      show12HourFormat: AvailabilityTimeFormatPrefs.use12HourFormat(),
     };
   }
 
   render() {
     const { availabilities, timezone, timeout } = this.props;
+    const { show12HourFormat } = this.state;
 
-    const tableContent =  _.map(availabilities, ({ day, start_time, end_time, id }) => {
+    const tableContent =  _.map(availabilities, ({ day, start_time, end_time, id, start_time_12_hour, end_time_12_hour  }) => {
```

Comment:
Are "start_time" and "end_time" still necessary after adding "start_time_12_hour" & "end_time_12_hour"?

Might be worth consolidating in the decorator and across components, to avoid future confusion.

Reply:
These are needed to allow users to toggle back and forth between 12-hour and 24-hour time formats, because they contain the original 24 hour time. We could change end_time to end_time_24_hour and start_time to start_time_24_hour.